src/main/java/org/truffleruby/core/regexp/MatchDataNodes.java

```diff
@@ -282,14 +281,59 @@ protected boolean inBounds(DynamicObject matchData, int index) {
     }
 
 
+    public abstract static class ValuesNode extends CoreMethodArrayArgumentsNode {
+
+        @Child private RopeNodes.MakeSubstringNode makeSubstringNode = RopeNodes.MakeSubstringNode.create();
+
+        public static ValuesNode create() {
+            return ValuesNodeFactory.create(new RubyNode[0]);
+        }
+
+        public abstract Object[] execute(DynamicObject matchData);
+
+        @Specialization(guards = "hasValues(matchData)")
```

Comment:
I was wondering if a `@Cache` would make sense here?

Reply:
See the discussion of MRI behaviour above. We should recreate these values every time so the point will be moot real soon. :grin: